Automattic/woocommerce-payments
# Fix overview screen type error when the account has no deposit scheduled

Fixes #5863

#### Changes proposed in this Pull Request

This PR fixes the error `TypeError` that occurs on the Payments → Overview screen when an account has no scheduled upcoming deposit.

The `nextScheduled`, `lastPaid` and `instant` properties of the type `AccountOverview.Overview`  have been changed to optional, since they can be returned as `undefined` in some cases.

https://github.com/Automattic/woocommerce-payments/blob/279046a35872335d77bea203c9b5b3085aa198a5/client/types/account-overview.d.ts#L43-L50

I've also made the `overview` prop of the `getNextDeposit()` function optional, which can be `undefined` if `useAllDepositsOverviews` has not resolved the data yet. This was throwing another error once the error above was fixed.

https://github.com/Automattic/woocommerce-payments/blob/279046a35872335d77bea203c9b5b3085aa198a5/client/components/deposits-overview/utils.ts#L21-L23

#### Testing instructions





1. Onboard to WCPay with a new account.
2. Enable the `_wcpay_feature_simplify_deposits_ui` feature flag: Run `update_option( '_wcpay_feature_simplify_deposits_ui', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/).
3. View the Payments → Overview screen.
4. Ensure no error is thrown in the console.
5. Ensure that the currency shown in the estimated deposits table matches the selected/default currency.

<img width="716" alt="image" src="https://user-images.githubusercontent.com/3147296/227843326-b70440ad-586c-47ed-bcc8-01cbf72f5787.png">


-------------------


**Post merge**
